File: services/api/app/schemas/ai.py
from datetime import datetime
from typing import Any, Literal

from pydantic import BaseModel


class AIDetectionResponse(BaseModel):
    modality: Literal["image", "text"]
    provider: str
    model: str
    provider_version: str
    score: float | None
    label: Literal["likely_ai", "uncertain", "likely_human", "unavailable"]
    calibrated: bool
    cached: bool
    latency_ms: int | None
    evaluated_at: datetime
    thresholds: dict[str, float]
    # Evidence level implied by the initial rules (docs/07): PROBABLE / POSSIBLE / UNKNOWN.
    evidence_level: Literal["PROBABLE", "POSSIBLE", "UNKNOWN"]
    raw: dict[str, Any]
    limitations: list[str]

File: packages/shared-types/src/ai.ts
/** Mirrors `app/schemas/ai.py`. A detector result is a probabilistic signal, never proof. */

export type AILabel = "likely_ai" | "uncertain" | "likely_human" | "unavailable";

export interface AIDetectionResponse {
  modality: "image" | "text";
  provider: string;
  model: string;
  provider_version: string;
  /** 0 = no AI signal … 1 = strongest, as reported by the provider. Not calibrated. */
  score: number | null;
  label: AILabel;
  calibrated: boolean;
  cached: boolean;
  latency_ms: number | null;
  evaluated_at: string;
  thresholds: { high: number; medium: number };
  evidence_level: "PROBABLE" | "POSSIBLE" | "UNKNOWN";
  raw: Record<string, unknown>;
  limitations: string[];
}
